File: scanner/actions/HomeAssistantAction.js
const fetch = require('node-fetch')
const Action = require('./Action')
const https = require('https')

class HomeAssistantAction extends Action {
  process() {
    this.request(this.payload())
  }

  payload() {
    const envVars = this.envVars()
    const payload = {}

    Object.keys(envVars).forEach(key => {
      if (key.includes('HOMEASSISTANT')) {
        return
      }
      payload[key.toLowerCase()] = envVars[key]
    })

    return payload
  }

  async request(payload) {
    const protocol = this.config.ssl ? 'https://' : 'http://'
    const baseURL = `${protocol}${this.config.host}:${
      this.config.port
    }/api/events/magic_card_scanned`

    let headers = {'Content-Type': 'application/json'}
    if (this.config.token) {
      headers['Authorization'] = `Bearer ${this.config.token}`
    } else if (this.config.password) {
      headers['x-ha-access'] = this.config.password
    }

    const init = {
      method: 'POST',
      headers: headers,
      body: JSON.stringify(payload),
    }

    // Compare to false so that we don't disable SSL if option omitted
    if (this.config.verify_ssl === false) {
      init.agent = new https.Agent({
        rejectUnauthorized: false,
      })
    }

    return fetch(baseURL, init)
      .then(res => res.text())
      .then(body => console.log(body))
      .catch(error => console.log(error))
  }
}

module.exports = HomeAssistantAction

File: scanner/actions/homeassistant_action.py
import requests
import traceback
from .action import Action


class HomeAssistantAction(Action):
    def process(self):
        self.request(self.payload())

    def payload(self):
        env_vars = self.env_vars()
        payload = {}
        for key, value in env_vars.items():
            if "HOMEASSISTANT" in key:
                continue
            payload[key.lower()] = value

        return payload

    def request(self, payload):
        protocol = "https://" if self.config.get("ssl", False) else "http://"
        base_url = "{}{}:{}/api/events/magic_card_scanned".format(
            protocol, self.config["host"], self.config["port"]
        )

        headers = {"Content-Type": "application/json"}
        if self.config.get("token", ""):
            headers["Authorization"] = "Bearer {}".format(self.config["token"])
        elif self.config.get("password", ""):
            headers["x-ha-access"] = self.config["password"]

        verify = True
        if self.config.get("verify_ssl", None) is False:
            verify = False

        try:
            resp = requests.post(base_url, data=payload, headers=headers, verify=verify)
            resp.raise_for_status()
        except Exception:
            traceback.print_exc()
        else:
            return resp.json()
